stock_location_flowable/models/stock_location.py

```diff
@@ -0,0 +1,219 @@
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import json
+
+
+class Location(models.Model):
+    _inherit = "stock.location"
+
+    flowable_storage = fields.Boolean()
+    flowable_mix_blocked = fields.Boolean(compute="_compute_flowable_mix_blocked")
+    flowable_capacity = fields.Float()
+    flowable_uom_id = fields.Many2one(comodel_name="uom.uom", string="Unit of Measure")
+    flowable_sequence_id = fields.Many2one(comodel_name="ir.sequence")
+    flowable_allowed_product_ids = fields.Many2many(
+        comodel_name="product.product", string="Allowed Products"
+    )
+    flowable_mixing_production_id = fields.Many2one(
+        comodel_name="mrp.production",
+        compute="_compute_flowable_mixing_production_id",
+        store=True,
+    )
+    flowable_capacity_occupied = fields.Float(
+        compute="_compute_flowable_capacity_occupied"
+    )
+    # Mirar el tema de perque la string esta " " mira el percentpie
+    flowable_percentage_occupied = fields.Float(
+        compute="_compute_flowable_percentage_occupied"
+    )
+    use_new_order_lot = fields.Boolean()
+    json_popover = fields.Char(
+        "JSON data for the popover widget", compute="_compute_json_popover"
+    )
+
+    def _compute_json_popover(self):
+        for rec in self:
+            rec.json_popover = json.dumps(
+                {
+                    "title": _("Flowable Location Warning"),
+                    "msg": _(
+                        "This location is blocked because it is a flowable"
+                        " and it has a manufacturing order assigned."
+                    ),
+                }
+            )
+
+    @api.depends("flowable_mixing_production_id")
+    def _compute_flowable_mix_blocked(self):
+        for rec in self:
+            rec.flowable_mix_blocked = (
+                True
+                if rec.flowable_storage and rec.flowable_mixing_production_id
+                else False
+            )
+
+    @api.depends(
+        "usage",
+        "flowable_storage",
+        "quant_ids.product_id.stock_move_ids.move_line_ids.state",
+    )
+    def _compute_flowable_mixing_production_id(self):
+        for rec in self:
+            # En compra funciona perque encara no esta enllaçat amb la location
+            # En MO no permet assignar una move_line_id amb la location perque
+            # genera el moviment al stock.location (es confos)
+            if rec.flowable_storage and rec.usage != "view":
+                production = (
+                    rec.env["stock.move.line"]
+                    .search(
+                        [
+                            "|",
+                            ("location_dest_id", "=", rec.id),
+                            ("location_id", "=", rec.id),
+                            ("state", "not in", ("draft", "cancel", "done")),
+                            ("company_id", "=", rec.env.user.company_id.id),
```

Comment:
It should be the company of the location, but what if the location has no company? Falling back to user current company?